src/classes/CON_DoNotContact_TEST.cls

```diff
@@ -29,255 +29,1346 @@
+            System.assertEquals(true, con.Exclude_from_Household_Formal_Greeting__c, 'Contact should be excluded from Household Formal Greeting.');
+            System.assertEquals(true, con.Exclude_from_Household_Informal_Greeting__c, 'Contact should be excluded from Household Informal Greeting.');
+            System.assertEquals(true, con.Naming_Exclusions__c.contains('Household__c.Name'), 'Naming Exclusions field should include Household Name.');
+            System.assertEquals(true, con.Naming_Exclusions__c.contains('Household__c.Formal_Greeting__c'), 'Naming Exclusions field should include Household Formal Greeting.');
+            System.assertEquals(true, con.Naming_Exclusions__c.contains('Household__c.Informal_Greeting__c'), 'Naming Exclusions field should include Household Informal Greeting.');
+        }
     }
 
     /**************************************************************************************************************************
-    * Tests Inserting a deceased Contact to verify DoNotCall, HasOptedOutOfEmail and HasOptedOutOfFax are set
+    * @description Test method to verify that syncNamingExclusionsOnContact removes all household naming exclusions   
+    * when a contact is marked as being not deceased.
     */
+    @isTest 
+    private static void syncNamingExclusionsNegative(){
+        List<Contact> testContactList = UTIL_UnitTestData_TEST.getMultipleTestContacts(5);
 
-    @isTest
-    private static void doNotContactOnInsert() {
+        for (Contact con : testContactList){
+            con.Deceased__c = false;
+        }
+
+        Test.startTest();
+        CON_DoNotContact_TDTM conDoNotContactTDTM = new CON_DoNotContact_TDTM();
+        conDoNotContactTDTM.syncNamingExclusionsOnContacts(testContactList);
+        Test.stopTest();
+
+        for (Contact con : testContactList){
+            System.assertEquals(false, con.Exclude_from_Household_Name__c, 'Contact should not be excluded from Household Naming.');
+            System.assertEquals(false, con.Exclude_from_Household_Formal_Greeting__c, 'Contact should not be excluded from Household Formal Greeting.');
+            System.assertEquals(false, con.Exclude_from_Household_Informal_Greeting__c, 'Contact should not be excluded from Household Informal Greeting.');
+            System.assertEquals(true, String.isBlank(con.Naming_Exclusions__c), 'Naming Exclusions field should be cleared.');
+        }
+    }
 
-        Contact contact = UTIL_UnitTestData_TEST.getContact();
-        contact.do_not_contact__c = true;
+    /**************************************************************************************************************************
+    * @description Test method to verify that syncNamingExclusionsOnContact removes all household naming exclusions   
+    * when a null value is passed for Deceased.
+    */
+    @isTest 
+    private static void syncNamingExclusionsNull(){
```

Comment:
This is a good example of a test scenario for the odd behavior with Salesforce checkboxes. Good catch and coverage here. I admittedly would not have caught its absence in this code review.